features/command_line_interface/stdin.feature

```diff
@@ -33,7 +33,7 @@ Feature: Reek reads from $stdin when no files are given
       """
 
   Scenario: syntax error causes the source to be ignored
-    When I pass "def incomplete" to reek
+    When I pass "=" to reek
```

Comment:
Parser 2.3.0.1 [blows up](https://travis-ci.org/troessner/reek/jobs/102731465) on parsing `def incomplete`:

```
lib/parser/diagnostic.rb:121:in `[]=': 1...1 out of range (RangeError)
```

I tried to make Parser raise the expected `Parser::SyntaxError` and `=` was the first thing that worked. I can try coming up with something less crypting that doesn’t make Parser blow up (but raise what we expect).

(We should probably guard Reek users from Parser blowing up, but that’s a task for another PR…)